### PR TITLE
[SYCL] Remove mentions of host device from in-tree LIT tests

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

**Comment by steffenlarsen** [comment]
This is ready for review. The removed tests will be moved to the test-suite in an upcoming PR.

File: sycl/test/basic_tests/reduction_ctor.cpp [hunk 1]
```diff
@@ -1,166 +0,0 @@
-// RUN: %clangxx -fsycl %s -o %t.out
-// RUN: %RUN_ON_HOST %t.out
-
-// This performs basic checks such as reduction creation, getIdentity() method,
-// and the combine() method of the aux class 'reducer'.
-
-#include "reduction_utils.hpp"
```
Comment (steffenlarsen) [inline sycl/test/basic_tests/reduction_ctor.cpp:7]: This seems to be the last file using `reduction_utils.hpp`. As such it should also be removed.

**Review by v-klochkov — DISMISSED** [review]
Ok for _ESIMD_. Thank you.

**Review by steffenlarsen — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]

File: sycl/test/regression/fsycl-host-compiler-win.cpp [hunk 2]
```diff
@@ -25,11 +25,13 @@ int main() {
 
   {
     buffer<int, 1> b(data, range<1>(3), {property::buffer::use_host_ptr()});
+    auto B = b.get_access<access::mode::write>();
     queue q;
     q.submit([&](handler &cgh) {
-      auto B = b.get_access<access::mode::write>(cgh);
-      cgh.parallel_for<class test>(range<1>(3), [=](id<1> idx) {
-        B[idx] = 1;
+      cgh.host_task([=]() {
```
Comment (v-klochkov) [inline sycl/test/regression/fsycl-host-compiler-win.cpp:31]: Does 'host_task' remains supported even though 'host' is being removed?

File: sycl/test/regression/fsycl-host-compiler-win.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // RUN: %clang_cl -fsycl -fsycl-host-compiler=cl -DDEFINE_CHECK -fsycl-host-compiler-options="-DDEFINE_CHECK /std:c++17" /Fe%t1.exe %s
-// RUN: %RUN_ON_HOST %t1.exe
+// RUN: %t1.exe
 // REQUIRES: system-windows
```
Comment (v-klochkov) [inline sycl/test/regression/fsycl-host-compiler-win.cpp:3]: I don't understand how the test fails in CI on Linux: https://github.com/intel/llvm/runs/8267871292?check_suite_focus=true. Can you please check?

File: sycl/test/regression/fsycl-host-compiler-win.cpp (hunk 2 above)
Comment (steffenlarsen) [reply sycl/test/regression/fsycl-host-compiler-win.cpp:31]: Yes! host_task does not have anything to do with the host device. However, these tests are arguably not independent from the presence of a device, so they should be moved to the test-suite.

**Comment by steffenlarsen** [comment]
fsycl-host-compiler tests were moved to the test suite in https://github.com/intel/llvm-test-suite/pull/1240

**Review by v-klochkov — DISMISSED** [review]

File: sycl/test/regression/fsycl-host-compiler-win.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply sycl/test/regression/fsycl-host-compiler-win.cpp]: `host_task` is executed on `host` and `host` != `host device`.

`host device` term only exists in SYCL 1.2.1 specification and it means the ability to execute SYCL Kernel Functions even if no OpenCL devices are available.  `host` term still exists in SYCL 2020 and `host_task` is related to it, not to `host device`

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**